ReleaseProcessAutomation/Jira/JiraEntrancePoint.cs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership.  rubicon licenses this file to you under 
+// the Apache License, Version 2.0 (the "License"); you may not use this 
+// file except in compliance with the License.  You may obtain a copy of the 
+// License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT 
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the 
+// License for the specific language governing permissions and limitations
+// under the License.
+//
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Security.Cryptography.X509Certificates;
+using System.Xml.Schema;
+using ReleaseProcessAutomation.Configuration.Data;
+using ReleaseProcessAutomation.Jira.ServiceFacadeImplementations;
+using ReleaseProcessAutomation.SemanticVersioning;
+using Serilog;
+using Spectre.Console;
+
+namespace ReleaseProcessAutomation.Jira;
+
+public interface IJiraEntrancePoint
+{
+  void CreateAndReleaseJiraVersion (SemanticVersion currentVersion, SemanticVersion nextVersion, bool squashUnreleased = false);
+
+  void CheckJiraCredentials (Credentials credentials);
+}
+
+public class JiraEntrancePoint : IJiraEntrancePoint
```

Comment:
I'm not sure that `CreateAndReleaseJiraVersion` should be made into a step as it more closely resembles the behaviour of `MsBuild`, which is not a step aswell. 

Making `CreateAndReleaseJiraVersion` a step would also create several problems it can be called in different parts of the application, therefore not always having a following step, as this would completely break up the order of how  things are done. 

I would suggest renaming the `JiraEntrancePoint` to `CreateAndReleaseJiraVersion` like you mentioned and renaming it's only public method to `Execute`.

Reply:
Yes looking at it again making it a step is probably not the best idea :D

We should split these two methods into two classes maybe `IJiraFunctionality` and `IJiraAuthenticator` or something. Since we are not converting the class to a step it might get new functionality in the future, which is why I would prefer to stay away from `CreateAndReleaseJiraVersion`.

Since we have some shared logic like `JiraUrlWithPostfix` we probably want to move that into another class like `JiraMetadata` and have the other ones depend on it for the functionality.